_posts/2022-03-10-workflow.md

```diff
@@ -0,0 +1,33 @@
+---
+
+<div style="text-align: justify">
+ 
+## Workflow
+
+A computational workflow combines data and it's opearations of the data. Several processes are coordinated to handel dependencies between different computational steps. Workflows looks at methods which are used for (some or all of) the following: collecting data, preparing, analysis and predictive modelling and simulation of data ([Gobel et al (2020)][1]). A tool or a model can be concidered a computational  workflow as they can support the analysis.
+ 
+![Workflow](https://user-images.githubusercontent.com/30128518/157625071-57a8a629-8319-4168-bc57-f92c4851b2ba.png)
+
+Creating a workflow can be divided into three stages.
+- First, you need to identify and describe the process and steps needed to characterise the relationship between the data   inputs and outputs. Mapping the required formats and data for potential project based database or visualisation is important to reach the desired final product.
+ - Second, when developing the workflow the scripts and codes that are produced needs to be easily found and also clearly marked which sequence they are to be executed. Here several options are possible to support, where in computational workflows [snakemake](https://snakemake.readthedocs.io/en/stable/) is one option which executes the required steps through a Snakefile until the final output is obtained. If the computations are in Python then a [Jupyter Notebook](https://jupyter.org/) (which allows for markdown comments) can be easy to describe the steps, execute functions and plot some of the results.
+ - Third, it is important to clearly defined the different steps so that it can easily be implmened with tests and checking that the workflow is working properly. This will help to identify bugs and issues that could arise.
+ 
+ ### Reproducible and FAIR
+ When developing computational workflows your work become more reusable and reproducible for others and this in turn can help increasing collaboration. Well defined workflows contribute to [FAIR data](https://www.go-fair.org/fair-principles/) as it provide tracability of the results and data used to produce an output. If the workflow is complex then also time and energy can be saved by creating a workflow.
+
+### Example of collaborative workflow
+ 
+The [PyPSA-Eur: An Open Optimisation Model of the European Transmission System](https://github.com/PyPSA/pypsa-eur) is a good example of how to work on collaborative workflows. The workflow is written in Python and executed using [snakemake](https://snakemake.readthedocs.io/en/stable/). Each release is deposited on Zenodo and ascribed a DOI. The documentation is available on [ReadTheDocs](https://pypsa.readthedocs.io/en/latest/) and there is also guidance that helps users identify if the available data is suitable for their needs, together with links to find more information. Furthermore, is each release documented in a academic short paper and archived as a pre-print in arxiv. 
```

Comment:
Here you mention that PyPSA releases are deposited on Zenodo. Could you maybe add above then a quick reference to releases in the context of workflows?

Reply:
Same for the documentation. Maybe in stage 2 of how to create a workflow, you could add an even simpler option of just collecting the documentation of the steps and processing work done in a document/.txt file with a predefined set of metadata/information that can guide in the workflow?

Reply:
Great, I added a section on both parts in the main text.